### ISSUE TITLE
[Server Request] Jedi Knight II – Jedi Outcast

### ISSUE THREAD

**Issue by nateberkopec** [GameServerManagers/LinuxGSM#1202]
Hey, I'm working on a Jedi Outcast server [here](https://github.com/nateberkopec/LinuxGSM/tree/jk2).

The server needs a few of the assets from the game itself - assets0.pk3, assets1.pk3, and assets2.pk3. I figure OK, I'll get them from Steam - but since there's no Linux *client*, it refuses to download:

```
ERROR! Failed to install app '6030' (Invalid platform)
```

Can you tell me where I've gone wrong and what to do next? 

connects #1251

**Comment by nateberkopec** [comment]
Uh-oh, I think I see already. There's not a separate Steam App for the server, like, say, Arma. 

Am I screwed?

**Comment by UltimateByte** [comment]
https://steamdb.info/search/?a=app&q=Jedi+Outcast

1) This game does not support Linux
https://steamdb.info/app/6030/
2) No server appid exists for the server.
3) Adding a new server to LGSM is not a simple as changing the appid of an existing one.
https://github.com/GameServerManagers/LinuxGSM/wiki/New-Server

**Comment by nateberkopec** [comment]
Just to be clear, because I'm not sure I was understood - a Linux server *does* exist for JKII, but it does not contain some asset files. Those asset files are available in the (Windows) Steam download. Is there anything I can do?

**Comment by nateberkopec** [comment]
Also, if you'd look at my fork, I'm not "just changing the appid of an existing" game.

**Comment by UltimateByte** [comment]
My bad for the 3) then, i didn't read your commits, was late, and most of the time an error like that has nothing to do with LGSM so... :o))
The only chance for you to get it working is probably to use Wine. But do you really want it ?
For now LGSM refuses to add support for a game through Wine, for ethical, and practical reasons (for users as well as the devs).
Or maybe, there is a Mac OS server version, that might be compatible with Linux. Since you seem active & reactive, i'll re-open this. May i let you investigate about it ? If you find a way to just start natively the server (even without lgsm, just and executable to start), i'll be glad to help you implementing it.

**Comment by nateberkopec** [comment]
@UltimateByte JKII has a Linux executable server, I don't need wine. I've bundled up the JKII Linux server and it is installed via [install_server_files.sh](https://github.com/nateberkopec/LinuxGSM/commit/67d6376284a367795d9e8ae5382c5b29a74c75b4#diff-27b56802aa331362f5ba1aa7eb6b1cb1R29). The problem is that the server needs 3-4 asset files from the base game, but I can't download the base game via Steam because I get the platform check error above.

So, to recap:

1. I have a JKII dedicated server executable.
2. The steam AppID for JKII is Windows/Mac/Steamplay only
3. SteamCMD seems to *only* want to try to download the platform-specific version of the game (which does not exist).

**Comment by UltimateByte** [comment]
I see, thanks for the clarification.

We can host missing files and make them downloaded upon install, if it has a chance to work, if steam refuses to make them downloaded onto a linux server.

**Comment by nateberkopec** [comment]
Yes, adding the files manually works perfectly (it's actually what the *official documentation* tells you to do). But those files are copyrighted, so redistributing them is a no-no. That's why I was trying to get the files from Steam instead.

**Comment by AlexCSilva** [comment]
Jedi Knight II and Jedi Knight Jedi Academy are pretty old and the server files may not be on steam's database. When you're playing multiplayer on these games it doesn't show server info on steam, nor can you add favourites in steam servers. I'm guessing it may not be as supported compared to source games.

**Comment by AlexCSilva** [comment]
Another note, instead of using the original jk2ded it's better to use jk2mv, that way you can run server on any version with +set mv_version 1.02(or 1.03 or 1.04)

**Comment by nateberkopec** [comment]
@AlexCSilva my PR already includes jk2mv. As stated above, I don't need the server files (those are freely available and redistributed online) but I need 3-4 asset files from the client (doesn't matter which operating system). I can only get those, legally, from Steam. All I need is someone to tell me how to disable the platform check that SteamCMD makes.

**Comment by UltimateByte** [comment]
We host other game server content as well when it's the only way to make it work; nobody ever came to blame us for contributing to those games' continuation :p
Can you link the official doc you're referring to?
A good workaround to remain 100% legit would be to download it from their official servers and maybe display a link to them upon download.

**Comment by dgibbs64** [comment]
This might help https://developer.valvesoftware.com/wiki/SteamCMD#Cross-Platform_Installation

**Comment by nateberkopec** [comment]
@dgibbs64 Exactly what I needed! Thanks!

**Comment by AlexCSilva** [comment]
Awesome progress

**Comment by nateberkopec** [comment]
It's all idTech3. Once I finish JKII, Jedi Academy should be a piece of cake.

**Comment by AlexCSilva** [comment]
Yeah it should. Soldier of Forturne II would be nice too except, no steam support

**Comment by AlexCSilva** [comment]
Hey @nateberkopec how did you go?

**Comment by nateberkopec** [comment]
I don't really have time to work on this right now. For anyone wanting to pick this up:

* Use Quake3 as a base (same engine)
* Use JK2MV as the base executable rather than the official server executable
* Use the SteamCMD trick @dgibbs64 to force downloading the *windows* version of the game.

**Comment by AlexCSilva** [comment]
Okay, I'll look into it.

**Comment by AlexCSilva** [comment]
Working on this now, just need to add function to download and extract JK2MV https://github.com/AlexCSilva/LinuxGSM/tree/jk2

**Comment by dgibbs64** [comment]
connects #1251
